arch/arm/src/armv7-a/cp15_cacheops.c

```diff
@@ -44,9 +44,19 @@ static inline uint32_t ilog2(uint32_t u)
   return i;
 }
 
-static inline uint32_t cp15_cache_get_info(uint32_t *sets, uint32_t *ways)
+static inline uint32_t cp15_cache_get_info(uint32_t *sets, uint32_t *ways,
+                                           bool icache)
 {
-  uint32_t ccsidr = CP15_GET(CCSIDR);
+  uint32_t ccsidr;
+  uint32_t csselr;
+
+  csselr = CP15_GET(CSSELR);
+
+  csselr = (csselr & ~0x01) | (icache & 0x01);
```

Comment:
Instead of hard-coding 0x01 please use NVIC_CSSELR_IND (strange, it's only defined to armv7-m, but it should to armv7-a as well, PTAL)

Reply:
Done